app/forms/register_user_email_form.rb

```diff
@@ -29,6 +29,15 @@ def email
     email_address&.email
   end
 
+  # Note: This may perform a lot of DNS requests...
```

Comment:
I'm not sure how slow/fast DNS lookups are, but maybe the EmailNormalizer class could cache the "is gmail" of each domain name in Redis? still would be worth memoizing here probably, but for overall performance if this is a hot path

Reply:
actually related ---- adding network requests to a hot path like this can be risky, have we considered feature-flagging this check so we coudl disable it quickly if we discover any issues after deploying?

Reply:
@zachmargolis I suspect the cost of the MX record lookup outweighs the benefit, particularly upon registration. If the user has a custom domain, they have other (easy) options to forward *@domain to a single address.

Removing the MX record lookup would also eliminate the problem with IDN domains. When the speed of the application is taken into consideration, along with protecting against worst case scenarios, I do not think it's prudent to include the lookup.

Reply:
Aren't we already performing this MX check with the existing suspended user check via `blocked_email_address` below? It would be good if we could memoize that somehow, but otherwise I wasn't expecting either (a) any more network requests than were already happening or (b) any meaningful changes to the `EmailNormalizer` class.